pkg/apis/triggers/v1alpha1/event_listener_types.go

```diff
@@ -28,18 +28,23 @@ var _ apis.Validatable = (*EventListener)(nil)
 
 // EventListenerSpec defines the desired state of the EventListener, represented by a list of Triggers.
 type EventListenerSpec struct {
-	ServiceAccountName string    `json:"serviceAccountName"`
-	Triggers           []Trigger `json:"triggers,omitempty"`
+	ServiceAccountName string                 `json:"serviceAccountName"`
+	Triggers           []EventListenerTrigger `json:"triggers,omitempty"`
 }
 
-// Trigger represents a connection between TriggerBinding and TriggerTemplate; TriggerBinding provides extracted
-// values for TriggerTemplate to then create resources from.
-//
-// +k8s:deepcopy-gen=true
-type Trigger struct {
-	TriggerValidate *TriggerValidate   `json:"validate,omitempty"`
-	TriggerBinding  TriggerBindingRef  `json:"binding"`
-	TriggerTemplate TriggerTemplateRef `json:"template"`
+// EventListenerTrigger represents a connection between TriggerBinding and TriggerTemplate;
+// TriggerBinding provides extracted values for TriggerTemplate to then create resources from.
+type EventListenerTrigger struct {
+	TriggerValidate    *TriggerValidate     `json:"validate,omitempty"`
+	Binding            EventListenerBinding `json:"binding"`
```

Comment:
Seems heterogeneous to have `templateRef.name` and `binding.bindingRef.name`. Could this be flattened one level?

Reply:
There needs to be one more layer for the TriggerBinding, because of the parameters being passed to the TriggerBinding at `binding.params`. I couldn't think of a good way to get around this asymmetry; do you have any ideas?

Flattening to one level makes it unclear if the parameters are passed to the TriggerBinding or TriggerTemplate:
```
bindingRef:
  name: pipeline-binding
templateRef:
  name: pipeline-template
params:
- name: message
  value: Hello from the Triggers EventListener!
```

Reply:
I was thinking 
```yaml
apiVersion: tekton.dev/v1alpha1
kind: EventListener
metadata:
  name: listener
spec:
  serviceAccountName: tekton-triggers-example-sa
  triggers:
  - binding:
      name: pipeline-binding
      params:
      - name: message
        value: Hello from the Triggers EventListener!
    template:
      name: pipeline-template
```